.gitignore

```diff
@@ -1,5 +0,0 @@
-/.venv/
```

Comment:
These ignores are here to avoid polluting and accidentally committing invalid things

Reply:
I know - but overwrites already existing local ones. Your excludes are not other people excludes :-)

Reply:
I don't follow? This is scoped to the repository?  If anyone was to checkout the project and run `make build` for example they would have `buiild/` and `dist/` folders showing as a change with git and would be confusing.

Reply:
I know that, anyway I took now more than 4 hours to try to get it work pypi for both branch for a simple python script file without a result, that was the reason why I used to think of pypi as a single simple python as an overkill

Reply:
Why would your git repo contain a different `.gitignore`? Normally these kind of things are scoped to the project so everyone has the same setup. With the exception of global ones which feel like a bit like an anti-pattern to me.

What were/are you struggling with? 

My rational for this is to ease install and upgrades. Either with pure `pip` or by leveraging `pipx` 

e.g. with pipx it would be `pipx install decode-config` and `pipx upgrade decode-config`

Reply:
I have .gitignore in ..gitignore, nobody is interesting on my local excludes and don't want to tell anyone what their .gitignore should look like.
 do not see advantage for non-dveloper, anyway got not solve issue, that pypi I created in testpypi did not include decode-config.py itself so didnt work on a virgin system. In the end, pypi repo was annoying that I couldn't reuse a deleted package. sorry, i don't have the time to spend with it.

Reply:
ever tried PR for both branch?

Reply:
Those ignores one level up sounds like something that could live in your global gitignore config

See: https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/

Why only cater for the non-developer? I don't see what the issue is with what I provided? I've used the same thing on a few pypi packages and it just worked by passing a token.

Should be a case of just creating a pypi account and setting a GitHub secret with the token and on release publish on GitHub it should just work.

If you go down this route you may want to rotate the initial key to something scoped to the project for increased security.

Reply:
needs some small changes, will be integrated in next master release

Reply:
@inverse how to make trigger action also when I update an existing release? Just updated 12.3.0 to 12.3.1 but action was not triggered

Reply:
Hi @curzon01,

What are you trying to achieve when updating an existing release?

Wwhen you publish the release it'll upload the package. The problem would be when you update the package it'll raise a 400 HTTP error since the package is already published to pypi. 

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

Here are the other states that you can trigger for releases.

Reply:
I know the result 400 reasons.
Issue is that the action uses outdated release info after updating a release to create a package.
I updated the release by selecting a new tag and update the desc and the action creates a package which is not aligned with the updated release settings.

Is this solvable so after an release update the action creates a package from the updated release?

Reply:
To understand the current action status:
The last failed action basically is a result of the updated github release.
I updated release v12.3.0 to v12.3.1 which causes the error creating of the pypi packages using 12.3.0 which already exist.

cause github release update action didn't work, I downgraded release v12.3.1 to v12.3.0 and created a new release v12.3.1.

Reply:
From what I understand the action is only invoked when the release is published and is based off the new existing tag that it was created on.

so if you had a draft release and updated it, as soon as you publish it that's what it will be based on.

I'm not sure what happens if you update an already published release though - is that what the problem came from updating an already published release?

Reply:
exactly, when you update an already published release (that's possible), then the github source file packages are updated but the action access an outdated tag

Reply:
Trying to reproduce the issue on a new [repo](https://github.com/inverse/action-release-test)

- created release (CI kicked off)
- Update release (name + tag) - CI not kicked off

How did you get yours working?

Reply:
same steps as yours but different yml (publish.yml from this repo) from your Init PR - that's why I'm asking